pkg/tui/components/messages/messages.go

```diff
@@ -31,12 +31,15 @@ type Model interface {
 	AddErrorMessage(content string) tea.Cmd
 	AddAssistantMessage() tea.Cmd
 	AddSeparatorMessage() tea.Cmd
+	AddCancelledMessage() tea.Cmd
 	AddOrUpdateToolCall(agentName string, toolCall tools.ToolCall, toolDef tools.Tool, status types.ToolStatus) tea.Cmd
 	AddToolResult(msg *runtime.ToolCallResponseEvent, status types.ToolStatus) tea.Cmd
 	AppendToLastMessage(agentName string, messageType types.MessageType, content string) tea.Cmd
 	ScrollToBottom() tea.Cmd
 	AddShellOutputMessage(content string) tea.Cmd
 	AddSystemMessage(content string) tea.Cmd
+	RemoveSpinner()
```

Comment:
Instead of making this public it would be nicer maybe to have a command that is sent down to all the components that tell them that the stream is cancelled, WDYT? Just asking